ccm_web/server/src/invitation/invitation.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@nestjs/common'
+import { ConfigService } from '@nestjs/config'
+import { Config } from '../config'
+import FormData from 'form-data'
+import { randomUUID } from 'crypto'
+import axios from 'axios'
+import { InvitationAPIError } from './invitation.errors'
+import { CanvasUser } from '../canvas/canvas.interfaces'
+
+// Invitation service based on Cirrus
+
+@Injectable()
+export class InvitationService {
```

Comment:
Extra credit: What if this was `CirrusInvitationService`? I guess I was thinking that we could have the class implement an `InvitationService` interface, and then have it dynamically load the correct module based on what provider is configured. That's probably too fancy at this point, but one of the big hurdles I see to this tool being useful to other institutions is that they either don't have a service like Cirrus or they have another one.

Reply:
I can change the symbols to have a `Cirrus…` prefix.  However, in the interest of time, I think having it implement an abstract `InvitationService` interface should be a new issue to be addressed in the (near) future.

Reply:
Totally fine, that's why it was labeled "extra credit". I think adding `Cirrus` is nice.

Bigger priority is adding role validation.

Reply:
Changed and opened issue #295.